src/RemoteRequest/CachedRemoteGetRequest.php

```diff
@@ -97,13 +119,24 @@ public function get( $url ) {
 		}
 
 		if ( false === $cached_data || $cached_data->is_expired() ) {
-			$response    = $this->remote_request->get( $url );
-			$expiry      = $this->get_expiry_time( $response );
-			$cached_data = new CachedData( $response->getBody(), $expiry );
+			try {
+				$response = $this->remote_request->get( $url );
+				$status   = $response->getStatusCode();
+				$expiry   = $this->get_expiry_time( $response );
+				$headers  = $response->getHeaders();
+				$body     = $response->getBody();
+			} catch ( FailedToGetFromRemoteUrl $exception ) {
+				$status = $exception->getStatusCode();
+				$expiry = new DateTimeImmutable( "+ {$this->min_expiry} seconds" );
+				$body   = $exception->getMessage();
+			}
```

Comment:
@schlessera FYI, I added this in 00036be when testing whether caching worked as expected in the failure case, and I found it didn't. Please advise if anything looks off.

Reply:
I would hardcode `$status` in that case to `500`. Otherwise, you might have an exception status between 200 and 299 and the subsequent code would then consider it a successful request.

Reply:
Will it? Won't it only get thrown if the status code is not in the 2xx range?

https://github.com/ampproject/amp-wp/blob/182634c9363b897fe154c0f9db1b0f2b18fe88f5/src/RemoteRequest/WpHttpRemoteGetRequest.php#L130-L136

Reply:
Ah, yes, indeed. Alright then, looks good to me otherwise.